devlib/utils/parse_aep.py

```diff
@@ -0,0 +1,528 @@
+#!/usr/bin/env python
+#    Copyright 2018 Linaro Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import getopt
+import subprocess
+import signal
+import serial
+import time
+import math
+
+class AEP_parser(object):
+    prepared = False
+
+    def prepare(self, infile, outfile, summaryfile, verbose = False):
+        self.verbose = verbose
+
+        try:
+            self.fi = open(infile, "r")
+        except IOError:
+            print "WARN: Unable to open input file %s" % (infile)
+            print "Usage: parse_arp.py -i <inputfile> [-o <outputfile>]"
+            sys.exit(2)
+
+        self.parse = True
+        if len(outfile) > 0:
+            try:
+                self.fo = open(outfile, "w")
+            except IOError:
+                print "WARN: Unable to create %s" % (outfile)
+                self.parse = False
+        else:
+                self.parse = False
+
+        self.summary = True
+        if len(summaryfile) > 0:
+            try:
+                self.fs = open(summaryfile, "w")
+            except IOError:
+                print "WARN: Unable to create %s" % (summaryfile)
+                self.fs = sys.stdout
+        else:
+                self.fs = sys.stdout
+
+        self.prepared = True
+
+
+    def unprepare(self):
+        if not self.prepared:
+            # nothing has been prepared
+            return
+
+        self.fi.close()
+
+        if self.parse:
+            self.fo.close()
+
+        self.prepared = False
+
+
+    def __del__(self):
+        self.unprepare()
+
+
+    @staticmethod
```

Comment:
Should these statics just be internal utility function? If you want to keep them as as statics, please move them to above the `__init__`. 

Also, please move the `__del__` to below the regular methods. In general the order of class members should be:

1. class attributes
2. static/class methods
3. `__init__`
4. regular methods
5. internal (single leading underscore) methods
6. "special" (surrounded by double underscores) methods.